MarsRover/Rover.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+
+namespace MarsRover
+{
+    public class Rover
+    {
+        // has a Direction & a Location 
+        // Move - Forward/Backward
+        // Turn - Left/Right
+        public Rover(Direction direction, int x, int y)
+        {
+            Direction = direction;
+            Location = new Location(x, y);
+        }
+
+        public Direction Direction { get; private set; }
+
+        public Location Location { get; private set; }
+
+        public Direction Turn(Command command) // TODO: talk at Mentor meeting about changing to switch expression
+        {
+            var turnLeft = command == Command.TurnLeft;
+            var turnRight = command == Command.TurnRight;
+            
+            var north = Direction == Direction.North;
+            var south = Direction == Direction.South;
+
+            var turnWest = Direction.West;
+            if ((turnLeft && north) || (turnRight && south)) return turnWest;
+
+            var turnEast = Direction.East;
+            if ((turnRight && north) || (turnLeft && south)) return turnEast;
+
+            var west = Direction == Direction.West;
+            var east = Direction == Direction.East;
+
+            var turnSouth = Direction.South;
+            if ((turnLeft && west) || (turnRight && east)) return turnSouth;
+
+            var turnNorth = Direction.North;
+            if ((turnRight && west) || (turnLeft && east)) return turnNorth;
+
+            return Direction.None;
+        }
+
+        public Location Move(Command command)// TODO: does Rover need to know the Map dimensions?? // TODO: refactor

```

Comment:
Is this actually moving or is it just providing a `Location` to move to? Both are okay, but the name and what it actually does isn't matching up